jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java

```diff
@@ -90,230 +84,288 @@
         switch (prop.toLowerCase().replaceAll(" ", "")) {
             case "sauce":
             case "saucelabs":
-                DRIVER.capabilities.common = sauceCapabilities();
+                getJDISettings().DRIVER.capabilities.common = sauceCapabilities();
                 return sauceLabs();
-            case "browserstack": return browserstack();
-            default: return seleniumLocalhost();
+            case "browserstack":
+                return browserstack();
+            default:
+                return seleniumLocalhost();
         }
     }
-    private static Pair<String, JFunc1<String, String>> getSmartSearchFunc(String name) {
+
+    private Pair<String, JFunc1<String, String>> getSmartSearchFunc(String name) {
         if (!SMART_MAP_NAME_TO_LOCATOR.keys().contains(name)) {
             throw exception("Unknown JDISettings.ELEMENT.smartName: '%s'. Please correct value 'smart.locators.toName' in test.properties." +
-                "Available names: [%s]", name, print(SMART_MAP_NAME_TO_LOCATOR.keys()));
+                    "Available names: [%s]", name, print(SMART_MAP_NAME_TO_LOCATOR.keys()));
         }
         return Pair.$(name, SMART_MAP_NAME_TO_LOCATOR.get(name));
     }
-    private static UseSmartSearch getSmartSearchUse(String prop) {
+
+    private UseSmartSearch getSmartSearchUse(String prop) {
         String propLower = prop.toLowerCase().trim().replaceAll(" ", "");
         switch (propLower) {
-            case "false": return FALSE;
-            case "onlyui": return ONLY_UI;
-            case "uiandelements": return UI_AND_ELEMENTS;
-            case "always": return ALWAYS;
-            default: return UI_AND_ELEMENTS;
+            case "false":
+                return FALSE;
+            case "onlyui":
+                return ONLY_UI;
+            case "uiandelements":
+                return UI_AND_ELEMENTS;
+            case "always":
+                return ALWAYS;
+            default:
+                return UI_AND_ELEMENTS;
         }
     }
 
-    public static void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
+    public void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
         String path = "";
         try {
             path = System.getProperty(property, getProperty(property));
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
         if (isEmpty(path))
             path = defaultPath;
         Properties properties = new PropReader(path).getProperties();
         if (properties.isEmpty())
             return;
         try {
             setCapabilities.execute(properties);
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
     }
 
-    private static void setSearchStrategy(String p) {
-        p = p.toLowerCase();
-        if (p.equals("soft"))
+    private void setSearchStrategy(String param) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'setSearchStrategy(String)' has an NPath complexity of 580](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)